gwpy/timeseries/timeseries.py

```diff
@@ -1428,14 +1428,15 @@ def taper(self, side='leftright'):
         out *= planck(out.size, nleft=nleft, nright=nright)
         return out
 
-    def whiten(self, fftlength, overlap=0, method='scipy-welch',
-               window='hanning', detrend='constant', asd=None, **kwargs):
-        """White this `TimeSeries` against its own ASD
+    def whiten(self, fftlength=2, overlap=0, method='scipy-welch',
+               window='hanning', detrend='constant', asd=None,
+               fir=False, filter_duration=1, highpass=0, **kwargs):
```

Comment:
is it reasonable to tie the `filter_duration` to `fftlength` and remove the former option? they seem redundant to me, but I understand they are currently used in separate contexts.

Reply:
I prefer to keep `fftlength` and `filter_duration` as separate options. In this new scheme, the former is a feature of the ASD calculation, while the latter is a feature of the actual filter that doesn't necessarily relate to `fftlength` at all. This is one reason why I've introduced a `FrequencySeries.interpolate` method, which is very similar to what [PyCBC](http://pycbc.org/pycbc/latest/html/_modules/pycbc/psd/estimate.html#interpolate) does.

Reply:
Ok, sounds reasonable.